lib/filesystem.nix

```diff
@@ -54,4 +54,34 @@
       dir + "/${name}"
   ) (builtins.readDir dir));
 
+  # findBySuffix: String -> Path -> [ Path ]
+  #
+  # Given a suffix and directory, return a filtered list containing the full
+  # path to all files matching the given suffix recursively.
+  #
```

Comment:
```suggestion
  #
  # NOTE: If the path parameter is a derivation output, the derivation must be built
  #       before evaluation can proceed. This is commonly called "import from derivation"
  #       or IFD. Unlike a normal dependency, this creates a sequential bottleneck.
  #       If possible, process such directories in a build script instead of the Nix language.
  #       Nixpkgs disallows import from derivation. If the path parameter is a local directory,
  #       this note does not apply.
  #
```

Reply:
It feels weird to me to include a note about IFD only for this specific function, but not for others like `cleanSource`, etc. There's nothing special about `findByPrefix` or `find` that makes it more prone to IFD.

Reply:
I agree (with the previous comment)

Reply:
I mentioned moving this note elsewhere in a different thread, so we can reference it here. Do you have any suggestions where to put it?

Reply:
Is there a section in the manual talking about IFD? If there's not, there should be. I think that would be a good place to put such things.